apps/aecore/src/aec_peers.erl

```diff
@@ -973,38 +967,23 @@ on_resolve_peer(SourceAddr, PeerInfo, IsTrusted, State) ->
             State#state{ hostnames = HostMap2 }
     end.
 
-%% Handles hostname resolution event.
+%% Handles timeout of hostname resolution event.
 -spec on_resolve_hostname(reference(), string(), state()) -> state().
-on_resolve_hostname(Ref, Host, State) ->
-    #state{ hostnames = HostMap } = State,
+on_resolve_hostname(Ref, Host, #state{ hostnames = HostMap } = State) ->
     case maps:find(Host, HostMap) of
         {ok, {Ref, RetryCount, PeerMap}} ->
-            case inet:getaddr(Host, inet) of
-                {error, nxdomain} ->
-                    epoch_sync:info("Failed to resolve hostname ~p", [Host]),
-                    HostData = {undefined, RetryCount + 1, PeerMap},
-                    HostMap2 = resolver_schedule(Host, HostData, HostMap),
-                    State#state{ hostnames = HostMap2 };
-                {ok, Addr} ->
-                    HostMap2 = maps:remove(Host, HostMap),
-                    State2 = State#state{ hostnames = HostMap2 },
-                    maps:fold(fun
-                        (_, {undefined, PeerInfo, IsTrusted}, S) ->
-                            Peer = peer(Addr, PeerInfo, IsTrusted),
-                            on_add_peer(Addr, Peer, S);
-                        (_, {SourceAddr, PeerInfo, IsTrusted}, S) ->
-                            Peer = peer(Addr, PeerInfo, IsTrusted),
-                            on_add_peer(SourceAddr, Peer, S)
-                    end, State2, PeerMap)
-            end;
+            [ async_add_peer(SourceAddr, PeerInfo, IsTrusted) || 
+                {SourceAddr, PeerInfo, IsTrusted} <- maps:values(PeerMap) ],
+            State#state{ hostnames = HostMap#{ Host => {undefined, RetryCount + 1, PeerMap}}};
         _ ->
             State
     end.
 
 %% Handles event adding a new peer to the pool.
 -spec on_add_peer(inet:ip_address(), peer(), state()) -> state().
 on_add_peer(SourceAddr, Peer, State0) ->
-    State = maybe_unblock(State0),
+    #state{ hostnames = HostMap } = State0,
+    State = maybe_unblock(State0#state{hostnames = maps:remove(Peer#peer.host, HostMap)}),
```

Comment:
Multiple peers could have the same hostname, wouldn't that cancel them to be added?

Reply:
This seems fine, the original code alo removed the host from the hostnames to be resolved when one of the resolves succeeded.